### DISCUSSION TITLE
Option to choose the audio output device

### DISCUSSION THREAD

**Discussion by ZestMyu** [ciderapp/Cider#1205] [Feature Request]
Requesting the option to select the audio device which Cider will output to.
Similar to how the same can be done in foobar2000
![image](https://user-images.githubusercontent.com/4792742/176086904-abe57352-df52-40fd-a213-6e616bbb381b.png)

Windows does have the ability to route audio for applications, but I noticed that this settings doesn't stick for Cider (and some other applications).
For example, I reboot/sleep the PC or if I simply close/reopen Cider, Cider auto will start getting routed to the default audio device again.
![image](https://user-images.githubusercontent.com/4792742/176086090-088b6953-5611-4469-841e-82f499286617.png)

**Comment by maikirakiwi** [comment]
I couldn’t change the device in Web Audio API for some stupid reasons.

**Reply by maikirakiwi** [reply]
Probably on hold for a long while.

**Comment by DecibillyJoel** [comment]
In theory there should be some solution to this, because I've implemented a very crude solution of my own via a plugin. It simply calls SoundVolumeView (a cmd tool created by NirSoft) to set Cider.exe's output device to what I've hardcoded it to, which I'm pretty sure is just a cli form of changing the setting shown here. I don't know how SoundVolumeView does it, but the fact that it can do it means there's probably some powershell or cmd command out there that can get and set these options.

**Reply by gopgop9000** [reply]
Is there any way you could share that workaround? I tried just using the SoundVolumeView's SetAppDefault command and that simply just broke the audio for me until restarting the app.

**Reply by DecibillyJoel** [reply]
@gopgop9000 I've made it available [here on GitHub](https://github.com/DecibillyJoel/Cider-Plugin-Set-Windows-Audio-Device) with instructions in the readme. Let me know if you need help getting it up and running

**Reply by gopgop9000** [reply]
Thank you very much, it's incredibly helpful!
